Sources/Core/Presenter/UploadManager.swift

```diff
@@ -26,21 +28,38 @@ public final class UploadManager {
     /// Add an uploading item to the manager.
     /// - Parameter item: an uploading item.
     public func add(item: UploadingItem) {
-        items.insert(item, at: 0)
+        if item.type == .image {
+            images.insert(item, at: 0)
+        } else {
+            files.insert(item, at: 0)
+        }
```

Comment:
I think using `switch` would be better here. We'll get support from the compiler when adding a new type.